src/table/hooks/use-snapshot.ts

```diff
@@ -35,10 +36,28 @@ const useSnapshot = ({ layout, viewService, model, rootElement }: UseSnapshotPro
       );
       snapshotLayout.snapshotData.content = {
         scrollLeft: viewService.scrollLeft,
+        rowsPerPage: viewService.rowsPerPage,
+        page: viewService.page,
       };
     }
     return snapshotLayout;
   });
+  useImperativeHandle(
+    () => ({
+      getViewState() {
+        const totalsPosition = getTotalPosition(layout);
+        const { visibleRowStartIndex = -1, visibleRowEndIndex = -1 } = findVisibleRows(rootElement, totalsPosition);
+        return {
+          scrollLeft: viewService.scrollLeft,
+          visibleTop: viewService.qTop + visibleRowStartIndex,
+          visibleHeight: visibleRowEndIndex < 0 ? 0 : visibleRowEndIndex - visibleRowStartIndex + 1,
+          rowsPerPage: viewService.rowsPerPage,
+          page: viewService.page,
+        };
+      },
```

Comment:
This looks like duplicate code. Can you break it out?

Reply:
Yes, it partly duplicate. Both onTakeSnapshot and getViewState need to return scrollLeft, rowsPerPage and page, but getViewState need visibleTop and visibleHeight for fetching correct data rows while onTakeSnapshot does not need them.

Reply:
Updated!